### ISSUE TITLE
Harvesters still getting stuck in bleed

### ISSUE THREAD

**Issue by StormWingDelta** [OpenRA/OpenRA#14752]
Been noticing this while testing my own changes I'm working on and the code for unsticking them doesn't seem to be doing it's job for reasons I don't understand.  Sometimes I'll see several harvesters all parked around an Ore/Gem Mine in the case of RA and also in a few rare cases around refineries.  Doesn't happen too often but when it does the things refuse to move to any other tile than the one they last tried to move to.  Not sure if it's the AI causing this or the Harvesters themselves but I'll see what I can dig up.  That said it happens most often with the AI. Maybe add some kind of failsafe to reroute them to other refineries and resource patches in an ever increasing radius until it finds a new spot it won't get stuck?

Edit:
Images:
https://i.imgur.com/820mfj2.png
https://i.imgur.com/Z785RNp.png
https://i.imgur.com/I7IrkxT.png
https://i.imgur.com/7FZftZC.png
https://i.imgur.com/hxyBwUG.png
https://i.imgur.com/FjN86OR.png

Basically we've got mostly harvesters bugging out either around the resource spawners or around the refineries.  There's also a couple of cases of the AI having issues with land and naval units when using both at the same time but that's a matter for another post.

Here's the test map I was using and the replay: https://drive.google.com/open?id=10yqdS0JgRYnIdOx8GLWkF9afithvfS8o  ,  https://drive.google.com/open?id=1sg8satxJgIMogHhV55xunoSTtvjfN8fO

**Comment by GSonderling** [comment]
We should be careful not to eliminate micro while fixing this. Improving the AI is fine, but the player should still manage the economy.

**Comment by MunWolf** [comment]
This is unexpected behaviour, you can micro by making them dig resources closer to your refinery but having them randomly get stuck or refuse to move is not adding anything to the game, there is no balance consideration here.

**Comment by robertschifferli** [comment]
Does anyone think an acceptable solution would be to implement an "Idle harvester" notification in the lower left corner of the UI as Warcraft 3/SC2 does with idle workers?

**Comment by avalach21** [comment]
Does anyone think an acceptable solution would be to implement an "Idle harvester" notification in the lower left corner of the UI as Warcraft 3/SC2 does with idle workers?

**Comment by reaperrr** [comment]
The very reason why they don't get un-stuck (or at least nudged out of the way by other harvs) is that internally, they are *not* considered idle. Besides, the AI is already coded to let idle harvesters search for a new resource patch (even map-wide, if the initial scans within the normal scan radius turn up nothing).

The issue in the last screenshot, for example, is probably that the harv on the ref's unloading spot is completely surrounded by actors, with no available paths to any resources. As a result, it stops moving and blocks the other harvesters.
The only way to fix this properly would be to make the pathfinding/movement code smart enough to not consider mobile actors as blockers during the initial pathfinding, and then nudge them out of the way while moving towards the resource, if necessary. That's easier said than done, though.

**Comment by GraionDilach** [comment]
>Besides, the AI is already coded to let idle harvesters search for a new resource patch (even map-wide, if the initial scans within the normal scan radius turn up nothing

This seems to be regressed somewhere though in this cycle. Once I mentioned (can't find a ticket so prolly so on IRC) that i have a harvester paradrop power which I had to disable for the AI, because that check led to the AIs trying to order harvesters still in the air. Apparently that's no longer the case though  because on the playtest version, I accidentally enabled these paradrops and the AI never did anything with the harvesters gathered this way.

**Comment by matjaeck** [comment]
Related issues:
#12817
#12576
#10565
#3089
